vendor/github.com/containernetworking/cni/pkg/invoke/raw_exec.go

```diff
@@ -31,30 +33,54 @@ type RawExec struct {
 
 func (e *RawExec) ExecPlugin(ctx context.Context, pluginPath string, stdinData []byte, environ []string) ([]byte, error) {
 	stdout := &bytes.Buffer{}
+	stderr := &bytes.Buffer{}
 	c := exec.CommandContext(ctx, pluginPath)
 	c.Env = environ
 	c.Stdin = bytes.NewBuffer(stdinData)
 	c.Stdout = stdout
-	c.Stderr = e.Stderr
-	if err := c.Run(); err != nil {
-		return nil, pluginErr(err, stdout.Bytes())
+	c.Stderr = stderr
+
+	// Retry the command on "text file busy" errors
+	for i := 0; i <= 5; i++ {
+		err := c.Run()
+
+		// Command succeeded
+		if err == nil {
+			break
+		}
+
+		// If the plugin is currently about to be written, then we wait a
+		// second and try it again
+		if strings.Contains(err.Error(), "text file busy") {
+			time.Sleep(time.Second)
+			continue
+		}
```

Comment:
is there a path in kube that hits this?

Reply:
@saschagrunert you've added this https://github.com/containernetworking/cni/pull/763, 
can you help me to answer this quesiton :smile: ?

Reply:
There was a setup race where the init container downloads the CNI Plugin binary (like Cilium) and the runtime immediately picks up the (not finished copying) executable on disk. This caused issues in multiple distributions on cluster bootstrap, so we added this feature of retrying for sake of laziness.